libs/labelbox/src/labelbox/schema/task.py

```diff
@@ -214,3 +227,188 @@ def get_task(client, task_id):
         task: Task = tasks[0]
         task._user = user
         return task
+
+
+class DataUpsertTask(Task):
+    """
+    Task class for data row upsert operations
+    """
+    __max_donwload_size: Final = MAX_DATAROW_PER_API_OPERATION
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._user = None
+
+    @property
+    def result(self) -> Optional[List[Dict[str, Any]]]:  # type: ignore
+        """
+        Fetches maximum 150K results. If you need to fetch more, use `result_all` property
+        """
+        if self.status == "FAILED":
+            raise ValueError(f"Job failed. Errors : {self.errors}")
+        return self._results_as_list()
+
+    @property
+    def errors(self) -> Optional[List[Dict[str, Any]]]:  # type: ignore
```

Comment:
I'm curious why we are not simply supporting a single `results` and `errors` method which returns a generator?

Reply:
because it will not be backward-compatible

Reply:
This is likely going to be confusing for users, i.e. `results` vs. `results_all`.

Returning a generator will likely not break things in practice.

Reply:
It will, I think, if someone is trying to use returns as a List